src/openqaoa-qiskit/backends/qaoa_qiskit_qpu.py

```diff
@@ -128,24 +131,38 @@ def qaoa_circuit(self, params: QAOAVariationalBaseParams) -> QuantumCircuit:
         qaoa_circuit: `QuantumCircuit`
             The final QAOA circuit after binding angles from variational parameters.
         """
+        parametric_circuit = deepcopy(self.parametric_circuit)
```

Comment:
This may be the part slowing down the code, perhaps.

Reply:
maybe, but we do need this one since we don't want to modify the original, already-constructed, self.parametric_circuit, right?

Reply:
we construct a new circuit after `bind_parameters`, why can't we append these instructions to this circuit instead of deepcopying the parametric circuit and then adding instructions to it? After this step I mean `circuit_with_angles = self.parametric_circuit.bind_parameters(memory_map)`